lib/Dialect/FIRRTL/Transforms/GrandCentralTaps.cpp

```diff
@@ -780,23 +794,33 @@ void GrandCentralTapsPass::runOnOperation() {
         // The code tries to come up with a relative path from the data tap
         // instance (path being the absolute path to that instance) to the
         // tapped thing (prefix being the path to the tapped port, wire, or
-        // memory) by calling stripCommonPrefix(prefix, path). Bug 2767 was
-        // caused because, the path in the NLA was not considered for this
-        // common prefix stripping (prefix is the path to the root of the NLA).
-        // This leads to overly pessimistic paths like
-        // Top.dut.submodule_1.bar.Memory[0], which should rather be
-        // DUT.submodule_1.bar.Memory[0]. To properly fix this, the path in the
-        // NLA should be inlined in the prefix such that it becomes part of the
-        // prefix stripping operation.
+        // memory) by calling stripCommonPrefix(prefix, path).  If the tapped
+        // thing includes an NLA, then the NLA path is appended to the rest of
+        // the path before the common prefix stripping is done.
 
         // Determine the shortest hierarchical prefix from this black box
         // instance to the tapped object.
-        Optional<InstancePath> shortestPrefix;
+        Optional<SmallVector<InstanceOp>> shortestPrefix;
```

Comment:
This change is to get around the fact that `InstancePath` is an `ArrayRef<InstanceOp>`. This is problematic as there is no existing array to point to once NLAs are considered. I'm not extremely happy about this implementation, but it stays as close as possible to the existing algorithm without completely reworking it.